pykanidm/kanidm/__init__.py

```diff
@@ -218,13 +256,24 @@ async def auth_begin(
             headers=self.session_header(),
         )
         if response.status_code != 200:
-            # TODO: write mocked test for this
+            # TODO: mock test for auth_begin raises AuthBeginFailed
             raise AuthBeginFailed(response.content)
 
         retobject = AuthBeginResponse.parse_obj(response.data)
         retobject.response = response
         return response
 
+    async def authenticate_with_token(self, token: str) -> ClientResponse:
```

Comment:
I don't think we should rely on the cookie here. I think it was there as a "legacy" thing, but the tokens are getting bigger and we'll hit the 4096 byte limit soon. So authenticate_with_token should just set self.token and then you should use bearer from then on? That makes more sense here I think that doing an exchange step.

Reply:
Yep, removing it as per our other discussions. The authentication will go through ... the other methods. :D